airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStreamCopier.java

```diff
@@ -27,24 +40,90 @@ public RedshiftStreamCopier(String stagingFolder,
                               SqlOperations sqlOperations) {
     super(stagingFolder, destSyncMode, schema, streamName, Strings.addRandomSuffix("", "", 3) + "_" + streamName, client, db, s3Config,
         nameTransformer, sqlOperations);
+    objectMapper = new ObjectMapper();
+  }
+
+  @Override
+  public void copyStagingFileToTemporaryTable() {
+    var possibleManifest = Optional.ofNullable(createManifest());
+    LOGGER.info("Starting copy to tmp table: {} in destination for stream: {}, schema: {}, .", tmpTableName, streamName, schemaName);
+    possibleManifest.stream()
+        .map(this::putManifest)
+        .forEach(this::executeCopy);
+    LOGGER.info("Copy to tmp table {} in destination for stream {} complete.", tmpTableName, streamName);
   }
 
   @Override
-  public void copyS3CsvFileIntoTable(JdbcDatabase database, String s3FileLocation, String schema, String tableName, S3Config s3Config)
-      throws SQLException {
+  public void copyS3CsvFileIntoTable(JdbcDatabase database, String s3FileLocation, String schema, String tableName, S3Config s3Config) {
+    throw new RuntimeException("Redshift Stream Copier should not copy individual files without use of a manifest");
+  }
```

Comment:
I know this breaks the initial abstraction, but it was necessary and I wanted to make this change more surgical rather than structural to all related s3 connectors.